src/sparseml/pytorch/sparsification/training/modifier_params.py

```diff
@@ -142,16 +144,23 @@ def update(
             (calculate batch number using this and epoch)
         """
         super().update(module, optimizer, epoch, steps_per_epoch)
+        self._check_update(epoch, steps_per_epoch)
 
-        if self.start_pending(epoch, steps_per_epoch):
-            self._original.clear()
+    def _enable(self, param: Parameter):
+        param.requires_grad_(True)
+
+    def _disable(self, param: Parameter):
+        param.requires_grad_(False)
+        param.grad = None  # clear to prevent optimizer updates
 
+    def _check_update(self, epoch: float, steps_per_epoch: int):
+        if self.start_pending(epoch, steps_per_epoch) and not self._original:
```

Comment:
I'm curious: do we actually need the `not self._original` condition?

Reply:
Yes we do for cases when modifier kicks in immediately during `def initialize(...)` (e.g. gradual pruning with `start_epoch=0`). `not self._original` prevents enabling/disabling again when `def update(...)` is called for the first time in the gradual run.